src/supplemental.test.js

```diff
@@ -0,0 +1,91 @@
+var {calcSingleHitSupplementalDamage, calcThirdHitSupplementalDamage} = require('supplemental.js');
```

Comment:
const ... and add empty line to next.

`calcSingleHitSupplementalDamage` names can be like `calcSingleHitDamage` ?
in global_logic.js, use with module name prefix can show the context calc what damage.
 `supplementa.calcSingleHitDamage`

this change require describe("#calcSingleHitSupplementalDamage" ...  labels change too.

Reply:
> I was going to suggest .bind() approach.
> 
> Unit tests and more ideas (coding topic)
> https://codepen.io/anon/pen/mZdbPN?editors=0011
> 
> * [ ]  separate to supplemental.js module
>   
>   * [ ]  add unit tests supplemental.test.js
> * [ ]  case "hp_based": fall-through to case "other"
> * [ ]  improve additionalVal (ES6)
> 
> NOTE: I did not check ES6 supports, at least it worked in codepen.io
> but I am not sure that with Motocal's babel configuration.
> 
> Noticed in unit tests
> 
> * switch/default case is rare to reach.
>   
>   * that does not report single typo in supplementalDamageArray

@L170 
```js
        let vals = calcSupplementalDamage(["unknown"], supplementalDamageArray, INITIAL_VALS());
```
this would require exporting the original function